### PR TITLE
Recipe app - best dishes V 1.0

### PR DESCRIPTION
**Pull request by Diegogagan2587**
# Implemented Features
_Team members:_ **2**
- A login page.
- A registration page.
- Home Page
- A food list (with all CRUD implementation, except for 'update').
- A recipe list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

_**Note for code reviewer:**_ Dear code reviewer, remember that if you want to review our project by branches you can open our [Closed Pull Request](https://github.com/Diegogagan2587/Recipe-app-bests-dishes/pulls?q=is%3Apr+is%3Aclosed), As well we have created a preview branch for deployment

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    return unless @recipe.nil?
+
+    flash[:error] = 'Recipe not found'
+    redirect_to recipes_path
+  end
+
+  def public_recipe
+    @recipes = Recipe.includes(:user).where(public: true).order(created_at: :desc)
+  end
```
Comment (Gambit142) [inline app/controllers/recipes_controller.rb:33]: - [ ] One of the requirements is that the public recipe should be displayed ordered by the newest in terms of the newest to be made public and not when the recipe was created. Kindly change the logic for this. 👍 😎

File: app/views/recipes/public_recipe.html.erb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+              </div>
+            <% end %>
+          </div>
+        <% else %>
+          <p class="text-center">You haven't created any recipes yet.</p>
+        <% end %>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (Gambit142) [inline app/views/recipes/public_recipe.html.erb:57]: - [ ] Based on the wireframe and project requirements, the **_Public Recipe page_** should contain the owner of the recipe, the total food items needed to prepare the recipe and the total cost of the recipe. Kindly include this on your page. 👍 😎 

### Wireframe

image

### Your Design

image

#### N.B: On the **_Public recipe page_**, there shouldn't be any button to add ingredient. Kindly implement this. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Diegogagan2587 @henask12 👋, 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- Good use of Gitflow ✔️ 
- Use of `devise` for authentication ✔️ 
- Use of `Rspec` as testing library ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/views/recipes/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                  <% end %>
+                </div>
+              </div>
+            </div>
+          <% end %>
+        </div>
+      <% end %>
+    </div>
+  </div>
+</div>
```
[the remark this answers is not in the document]
Comment (Diegogagan2587) [reply app/views/recipes/show.html.erb:58]: Dear @Gambit142 code reviewer, I'm confused at this part,  We are a team of 2 people, I was thinking that the shopping list is for teams of 3 people, Coul you please give us more info about this?

File: app/controllers/recipes_controller.rb (hunk 1 above)
Comment (Diegogagan2587) [reply app/controllers/recipes_controller.rb:33]: @henask12

File: app/controllers/foods_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @foods = current_user.foods
+  end
```
[the remark this answers is not in the document]
Comment (Diegogagan2587) [reply app/controllers/foods_controller.rb:5]: @Gambit142 I'm not sure if there is a problem here, I have created 4 differences foods and after checking the console in rails server I don't see any problem:
image

If I"'m wrong please let me know, i would appreciate further details!

**Review by Reem-lab — APPROVED** [review]
# Hi @Diegogagan2587,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ] Amazing working on the views, it is like the wireframe 🚀
- [ ] Good job working on the models and associations ✅
- [ ] Kudos for adding the schema with all the indexes on the foreign keys ⭐
- [ ]  Professional README 🤩
- [ ]  No Linters error ✅
- [ ] Deployed well 🚀
- [ ] Great routes for adding the needed ones ⭐

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**